### PR TITLE
[dtensor] Add single_dim_strategy functions for pooling, replicate-only, and histc ops

### PR DESCRIPTION
**Pull request by anshul-si**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #179203
* __->__ #179202
* #179201
* #179200
* #179068

Add single_dim_strategy implementations for:
- Category G: Pooling ops (batch/channel dim sharding, partial for avg pools)
- Category H: Replicate-only ops (SVD, QR, diagonal, tril, triu, upsample, etc.)
- Category L: Histc (partial output when min/max specified)

New registrations are commented out; old register_op_strategy kept active.

### CODE REVIEW

**Comment by anshul-si** [comment]
@claude review this pr

File: torch/distributed/tensor/_ops/_math_ops.py [hunk 1]
```diff
@@ -1047,33 +1033,43 @@ def linalg_replicate_strategy(op_schema: OpSchema) -> OpStrategy:
 ]
 
 
-@register_op_strategy(
+@register_single_dim_strategy(
     AVG_POOL_OPS + MAX_POOL_OPS,
     schema_info=RuntimeSchemaInfo(1),
+    allow_uneven_sharding=True,
```
Comment (pianpwk) [inline torch/distributed/tensor/_ops/_math_ops.py:1032]: nit: uneven sharding is ok for pooling? just double checking, but seems ok?

File: torch/distributed/tensor/_ops/_math_ops.py [hunk 2]
```diff
@@ -959,51 +957,39 @@ def powsum_single_dim_strategy(
...
+    aten.upsample_nearest2d.default,
+    aten.upsample_trilinear3d.default,
+    # TODO: support the full F.interpolate set of options.
+]
+
+
+@register_single_dim_strategy(
+    _REPLICATE_ONLY_OPS,
     schema_info=RuntimeSchemaInfo(1),
+    allow_uneven_sharding=True,
```
Comment (pianpwk) [inline torch/distributed/tensor/_ops/_math_ops.py]: a bit unnecessary

**Review by pianpwk — APPROVED** [review]

**Comment by pianpwk** [comment]
rest of the strategies look pretty 1-1 to non-single-dim version
